std/random.d

```diff
@@ -549,21 +551,67 @@ Parameters for the generator.
     enum size_t   shiftSize  = m; /// ditto
     enum size_t   maskBits   = r; /// ditto
     enum UIntType xorMask    = a; /// ditto
-    enum UIntType temperingU = u; /// ditto
+    enum size_t   temperingU = u; /// ditto
+    enum UIntType temperingD = d; /// ditto
     enum size_t   temperingS = s; /// ditto
     enum UIntType temperingB = b; /// ditto
     enum size_t   temperingT = t; /// ditto
     enum UIntType temperingC = c; /// ditto
     enum size_t   temperingL = l; /// ditto
+    enum UIntType initializationMultiplier = f; /// ditto
 
     /// Smallest generated value (0).
     enum UIntType min = 0;
     /// Largest generated value.
     enum UIntType max = UIntType.max >> (UIntType.sizeof * 8u - w);
-    static assert(a <= max && b <= max && c <= max);
+    // note, `max` also serves as a bitmask for the lowest `w` bits
+    static assert(a <= max && b <= max && c <= max && f <= max);
+
     /// The default seed value.
     enum UIntType defaultSeed = 5489u;
 
+    /// Bitmasks used in the 'twist' part of the algorithm
+    private enum UIntType lowerMask = (cast(UIntType) 1u << r) - 1;
+    private enum UIntType upperMask = (~lowerMask) & this.max; // ditto
+
+    /// Collection of all state variables
+    /// used by the generator
+    private struct State
+    {
+        /// State array of the generator.  This
+        /// is iterated through backwards (from
+        /// last element to first), providing a
+        /// few extra compiler optimizations by
+        /// comparison to the forward iteration
+        /// used in most implementations.
+        UIntType[n] data;
+
+        /// Cached copy of most recently updated
+        /// element of `data` state array, ready
+        /// to be tempered to generate next
+        /// `front` value
+        UIntType z;
+
+        /// Most recently generated random variate
+        UIntType front;
+
+        /// Index of the entry in the `data`
+        /// state array that will be twisted
+        /// in the next `popFront()` call
+        size_t index;
+    }
+
+    /// State variables used by the generator;
+    /// initialized to values equivalent to
+    /// explicitly seeding the generator with
+    /// `defaultSeed`
```

Comment:
@WebDrake they may confuse the reader (was this symbol supposed to be exposed? if not why is this a ddoc comment? etc). At any rate, it may be nice to transform these into block comments. Then we can switch from ddoc to regular comment with a single character change.